server.go

```diff
@@ -2527,8 +2558,16 @@ func (s *server) peerTerminationWatcher(p *peer, ready chan struct{}) {
 	// in question.
 	s.removePeer(p)
 
-	// Next, check to see if this is a persistent peer or not.
+	// If there were any notification requests for when this peer
```

Comment:
would say this should be moved up to just after acquiring the lock, otherwise we won't deliver notifications for teardowns that take the first conditional

Reply:
Fixed.